tests/generic_relations/tests.py

```diff
@@ -501,14 +500,36 @@ def test_query_content_type(self):
         with self.assertRaisesMessage(FieldError, msg):
             TaggedItem.objects.get(content_object="")
 
-    def test_unsaved_instance_on_generic_foreign_key(self):
-        """
-        Assigning an unsaved object to GenericForeignKey should raise an
-        exception on model.save().
-        """
+    def test_unsaved_instance_on_generic_foreign_key_save(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on save.
+        quartz = Mineral(name="Quartz", hardness=7)
+        tagged_item = TaggedItem(tag="shiny", content_object=quartz)
+        msg = (
+            "save() prohibited to prevent data loss due to unsaved related object "
+            "'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            tagged_item.save()
+
+    def test_unsaved_instance_on_generic_foreign_key_bulk_create(self):
+        # Assigning an unsaved object to a GFK should raise a ValueError on bulk create.
+        quartz = Mineral(name="Quartz", hardness=7)
+        tagged_item = TaggedItem(tag="shiny", content_object=quartz)
+        msg = (
+            "bulk_create() prohibited to prevent data loss due to unsaved related "
+            "object 'content_object'."
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            TaggedItem.objects.bulk_create([tagged_item])
+
+    def test_save_unsaved_instance_on_generic_foreign_key(self):
```

Comment:
```suggestion
    def test_save_generic_foreign_key_after_parent(self):
```

Reply:
Thank you - save unsaved was a really bad name :joy: